src/spdx/document_utils.py

```diff
@@ -29,3 +30,30 @@ def get_contained_spdx_elements(document: Document) -> Dict[str, Union[Package,
     contained_spdx_elements.update({snippet.spdx_id: snippet for snippet in document.snippets})
 
     return contained_spdx_elements
+
+
+def create_document_without_duplicates(document: Document) -> Document:
+    document_without_duplicates = deepcopy(document)
+    for elements in [
+        [document_without_duplicates.creation_info],
+        document_without_duplicates.files,
+        document_without_duplicates.packages,
+        document_without_duplicates.snippets,
+        document_without_duplicates.extracted_licensing_info,
+    ]:
+        for element in elements:
+            for key, value in element.__dict__.items():
+                if isinstance(value, list):
+                    value_without_duplicates = create_list_without_duplicates(value)
+                    setattr(element, key, value_without_duplicates)
+
+    return document_without_duplicates
+
+
+def create_list_without_duplicates(list_with_potential_duplicates: List[Any]) -> List[Any]:
+    list_without_duplicates = []
+    for element in list_with_potential_duplicates:
+        if element not in list_without_duplicates:
+            list_without_duplicates.append(deepcopy(element))
```

Comment:
I don't think we need to deepcopy the element here as it is already a copy of the original document.

Reply:
true! changed it.